contracts/modules/delegates/IexecPoco2Delegate.sol

```diff
@@ -25,574 +10,541 @@ import "./SignatureVerifier.sol";
-			// {
-			// 	// Check gas: https://ronan.eth.link/blog/ethereum-gas-dangers/
-			// 	assert(gasleft() > m_callbackgas / 63); // no need for safemath here
-			// }
-
-			// Pre solidity 0.6.0 version
-			(bool success, bytes memory returndata) = target.call{gas: m_callbackgas}(abi.encodeWithSignature("receiveResult(bytes32,bytes)", _taskid, _resultsCallback));
-			assert(gasleft() > m_callbackgas / 63);
-			// silent unused variable warning
-			success;
-			returndata;
-		}
-	}
-
-	/***************************************************************************
-	 *                            Array operations                             *
-	 ***************************************************************************/
-	function initializeArray(
-		bytes32[] calldata _dealid,
-		uint256[] calldata _idx)
-	external override returns (bool)
-	{
-		require(_dealid.length == _idx.length);
-		for (uint i = 0; i < _dealid.length; ++i)
-		{
-			initialize(_dealid[i], _idx[i]);
-		}
-		return true;
-	}
-
-	function claimArray(
-		bytes32[] calldata _taskid)
-	external override returns (bool)
-	{
-		for (uint i = 0; i < _taskid.length; ++i)
-		{
-			claim(_taskid[i]);
-		}
-		return true;
-	}
-
-	function initializeAndClaimArray(
-		bytes32[] calldata _dealid,
-		uint256[] calldata _idx)
-	external override returns (bool)
-	{
-		require(_dealid.length == _idx.length);
-		for (uint i = 0; i < _dealid.length; ++i)
-		{
-			claim(initialize(_dealid[i], _idx[i]));
-		}
-		return true;
-	}
+contract IexecPoco2Delegate is IexecPoco2, DelegateBase, IexecERC20Core, SignatureVerifier {
+    /***************************************************************************
+     *                    Escrow overhead for contribution                     *
+     ***************************************************************************/
+    function successWork(bytes32 _dealid, bytes32 _taskid) internal {
+        IexecLibCore_v5.Deal storage deal = m_deals[_dealid];
+
+        uint256 requesterstake = deal.app.price.add(deal.dataset.price).add(deal.workerpool.price);
+        uint256 poolstake = deal.workerpool.price.percentage(WORKERPOOL_STAKE_RATIO);
+
+        // seize requester funds
+        seize(deal.requester, requesterstake, _taskid);
+        // dapp reward
+        if (deal.app.price > 0) {
+            reward(deal.app.owner, deal.app.price, _taskid);
+        }
+        // data reward
+        if (deal.dataset.price > 0 && deal.dataset.pointer != address(0)) {
+            reward(deal.dataset.owner, deal.dataset.price, _taskid);
+        }
+        // unlock pool stake
+        unlock(deal.workerpool.owner, poolstake);
+        // pool reward performed by consensus manager
+
+        // Retrieve part of the kitty
+        uint256 kitty = m_frozens[KITTY_ADDRESS];
+        if (kitty > 0) {
+            kitty = kitty.percentage(KITTY_RATIO).max(KITTY_MIN).min(kitty); // fraction // at least this // but not more than available
```

Comment:
Could you please update this line (note: code will soon change though)?

Reply:
Done
[Fix comment](https://github.com/iExecBlockchainComputing/PoCo/pull/54/commits/7d5db24762238c322dc6755cfa52036dc80d956a)
Thanks.

Reply:
How about this?
[Fix kitty comment](https://github.com/iExecBlockchainComputing/PoCo/pull/54/commits/3915c9bb77cb9514803502797a7183c46ec7f30a)